mlflow/utils/file_utils.py

```diff
@@ -609,11 +612,13 @@ def get_or_create_nfs_tmp_dir():
 
     if is_in_databricks_runtime() and get_repl_id() is not None:
         # Note: In databricks, atexit hook does not work.
-        # The {nfs_root_dir}/repl_tmp_data/{repl_id} directory will be removed once databricks
-        # notebook detaches.
-        # The repl_tmp_data directory is designed to be used by all kinds of applications,
+        # The directory returned by `dbutils.entry_point.getReplNFSTempDir()`
+        # will be removed once databricks notebook detaches.
+        # The temp directory is designed to be used by all kinds of applications,
         # so create a child directory "mlflow" for storing mlflow temp data.
-        tmp_nfs_dir = os.path.join(nfs_root_dir, "repl_tmp_data", get_repl_id(), "mlflow")
+        repl_nfs_tmp_dir = _get_dbutils().entry_point.getReplNFSTempDir()
```

Comment:
Does this line always succeed on all Databricks runtimes?

Reply:
What would happen if this line is executed in DBR 10.4?

Reply:
I think so. (After this https://github.com/databricks/universe/pull/212893 merged)

I will confirm it with env team.

Reply:
> What would happen if this line is executed in DBR 10.4?

I think the change https://github.com/databricks/universe/pull/212893 will apply to all runtime versions , I will confirm with env team.

Reply:
The PR https://github.com/databricks/universe/pull/212893 is hard to backport (it adds new API), so I update mlflow code with fallback handling.